oci/layout/oci_transport.go

```diff
@@ -92,14 +74,11 @@ type ociReference struct {
 
 // ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an OCI ImageReference.
 func ParseReference(reference string) (types.ImageReference, error) {
-	var dir, image string
-	sep := strings.SplitN(reference, ":", 2)
-	dir = sep[0]
-
-	if len(sep) == 2 {
-		image = sep[1]
+	dir, image := oci.SplitDirAndImage(reference)
+	if image == nil {
+		return NewReference(dir, "")
 	}
-	return NewReference(dir, image)
+	return NewReference(dir, *image)
```

Comment:
I know. But as said, I wanted to keep the existing code behaving exactly how it was. And for that I need to differentiate between "" and nil. If you are ok to return "" and that an existing test gets changed, then yes.

Reply:
> I know. But as said, I wanted to keep the existing code behaving exactly how it was. And for that I need to differentiate between "" and nil.

All `ParseReference` does now is call `SplitDirAndImage`, and replace a `nil` with `""`. The difference is immediately erased.

Or are you talking (with the references to “existing test” about `ValidatePolicyConfigurationScope`)? That one does treat a trailing `:` and no `:` differently — but `ValidatePolicyConfigurationScope` should nowadays not expect a `:image` in the input at all, so that will disappear.